File: DS&Algo/SelectionSort.java
package udemyCourse.Sorting;

import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = { 5, 9, 3, 1, 2, 8, 4, 7, 6 };
        for (int i = 0; i < arr.length; i++) {
            int min = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[min]) {
                    min = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
        System.out.println("Array " + Arrays.toString(arr));
    }
}

File: CodeDump/SelectionSort.js
var a = [7,2,3,1,6,5,4];

for(var i = 0; i < a.length; i++) {
    var min = i;
    for(var j= min+1; j < a.length; j++) {
        if(a[j] < a[min]) {
            min = j; 
        }
    }
    var temp = a[i];
    a[i] = a[min];
    a[min] = temp;
}


console.log(a)
